code/modules/surgery/organs/autosurgeon.dm

```diff
@@ -81,24 +89,128 @@
 	starting_organ = /obj/item/organ/cyberimp/brain/anti_stun
 
-/obj/item/autosurgeon/syndicate/reviver
+/obj/item/autosurgeon/organ/syndicate/reviver
 	starting_organ = /obj/item/organ/cyberimp/chest/reviver
+
+/obj/item/autosurgeon/skillchip
+	name = "skillchip autosurgeon"
+	desc = "A device that automatically inserts a skillchip into the user's brain without the hassle of extensive surgery. \
+	 		It has a slot to insert a skillchip and a screwdriver slot for removing accidentally added items."
+	var/skillchip_type = /obj/item/skillchip
+	var/starting_skillchip
+	var/obj/item/skillchip/stored_skillchip
+
+/obj/item/autosurgeon/skillchip/syndicate
+	name = "suspicious skillchip autosurgeon"
+	icon_state = "syndicate_autoimplanter"
+
+/obj/item/autosurgeon/skillchip/Initialize(mapload)
+	. = ..()
+	if(starting_skillchip)
+		insert_skillchip(new starting_skillchip(src))
+
+/obj/item/autosurgeon/skillchip/proc/insert_skillchip(var/obj/item/skillchip/I)
+	if(!istype(I))
+		return
+	stored_skillchip = I
+	I.forceMove(src)
+	name = "[initial(name)] ([stored_skillchip.name])"
+
+/obj/item/autosurgeon/skillchip/attack_self(mob/living/carbon/user)//when the object it used...
+	if(!uses)
+		to_chat(user, "<span class='alert'>[src] has already been used. The tools are dull and won't reactivate.</span>")
+		return
+
+	if(!stored_skillchip)
+		to_chat(user, "<span class='alert'>[src] currently has no skillchip stored.</span>")
+		return
+
+	if(!istype(user))
+		to_chat(user, "<span class='alert'>[user]'s brain cannot accept skillchip implants.</span>")
+		return
+
+	if(!user.can_implant_skillchip(stored_skillchip))
+		var/fail_message = user.can_implant_skillchip_message(stored_skillchip)
+		to_chat(user, "<span class='alert'>[stored_skillchip] cannot be implanted. [fail_message]</span>")
+		return
+
+	if(user.implant_skillchip(stored_skillchip))
+		to_chat(user,"<span class='notice'>Operation complete!</span>")
+	else
+		to_chat(user,"<span class='notice'>Operation failed!</span>")
+		return
+
+	user.visible_message("<span class='notice'>[user] presses a button on [src], and you hear a short mechanical noise.</span>", "<span class='notice'>You feel a sharp sting as [src] plunges into your brain.</span>")
+	playsound(get_turf(user), 'sound/weapons/circsawhit.ogg', 50, TRUE)
+	stored_skillchip = null
+	name = initial(name)
+
+	if(uses != INFINITE)
+		uses--
+
+	if(!uses)
+		desc = "[initial(desc)] Looks like it's been used up."
+
+/obj/item/autosurgeon/skillchip/attackby(obj/item/skillchip/I, mob/user, params)
+	if(istype(I, skillchip_type))
+		if(stored_skillchip)
+			to_chat(user, "<span class='alert'>[src] already has a skillchip stored.</span>")
+			return
+		else if(!uses)
+			to_chat(user, "<span class='alert'>[src] has already been used up.</span>")
+			return
+		if(!user.transferItemToLoc(I, src))
+			return
+		stored_skillchip = I
+		to_chat(user, "<span class='notice'>You insert the [I] into [src].</span>")
+	else
+		return ..()
```

Comment:
```suggestion
/obj/item/autosurgeon/skillchip/attackby(obj/item/I, mob/user, params)
	if(!istype(I, skillchip_type))
		return ..()

	if(stored_skillchip)
		to_chat(user, "<span class='alert'>[src] already has a skillchip stored.</span>")
		return

	if(!uses)
		to_chat(user, "<span class='alert'>[src] has already been used up.</span>")
		return

	if(!user.transferItemToLoc(I, src))
		return

	stored_skillchip = I
	to_chat(user, "<span class='notice'>You insert the [I] into [src].</span>")
```
* Keep the argument's type consistent with their possible values, for clarity.
* Early returns.